### PR TITLE
`@remotion/studio`: Reload component when hook rules error occurs

### LINKED ISSUE #7398: Adding a second hook does not refresh the Studio anymore
This is only anecdotal, but I think that let's say you have a component

```tsx
const MyComp: React.FC = () => {
  return null;
}
```

and then you add a hook

```tsx
const MyComp: React.FC = () => {
  const frame = useCurrentFrame();
  return null;
}
```

There is a React error: "Rendered more hooks than before".
This is expected – but we intend to reload the Studio.
Same for when the order has changed and when there are fewer hooks now.

It seems to not work anymore recently.

We should also check if this applies to files:

- Where the component is not `export` ed
- Where the `<Composition>` is in the same file

Because we know this affects fast refresh.

Ideally we have some E2E tests that cover this.

### PR DESCRIPTION
**Pull request by JonnyBurger**
Adds an end-to-end regression test for the Studio fast-refresh case described in #7398: a component starts with one hook and then gains a second hook, which should force the Studio to recover instead of staying stale.

### What changed
- Adds a dedicated Studio fixture in `packages/example/src/HookOrderChangeE2e/` where the composition and component live in the same file and the component is not exported.
- Wires the fixture into the example Studio root so it can be visited from the existing e2e harness.
- Adds a Playwright test that edits the fixture from one hook to two hooks, waits for rebuild, and verifies the Studio shows the updated render.
- Extends the e2e setup/teardown helpers so the new fixture is safely restored between runs.

### Notes
- The test is intentionally structured to exercise the Fast Refresh edge case from the issue report, including the same-file/non-exported component scenario.
- No Studio runtime code changes were needed; the existing remount/recovery path is what the regression test now covers.

Closes #7398

### CODE REVIEW

File: packages/example/e2e/hook-order-change.test.mts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		expect(originalContent).toContain('one hook');
+		expect(originalContent).not.toContain('React.useState(false)');
+
+		const updatedContent = originalContent.replace(
+			'\tconst frame = useCurrentFrame();\n\n\treturn <div data-frame={frame}>one hook</div>;\n',
+			`\tconst frame = useCurrentFrame();\n\tconst [enabled] = React.useState(false);\n\n\treturn <div data-frame={frame} data-enabled={enabled ? 'yes' : 'no'}>two hooks</div>;\n`,
+		);
+		expect(updatedContent).not.toBe(originalContent);
+
+		const logCountBefore = readStudioLogs().length;
```
[the remark this answers is not in the document]
Comment (JonnyBurger) [reply packages/example/e2e/hook-order-change.test.mts]: Moved the log baseline to after navigation so the poll only watches the rebuild triggered by the file edit, not the initial build from `page.goto`.

File: packages/example/src/HookOrderChangeE2e/HookOrderChangeRepro.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (JonnyBurger) [reply packages/example/src/HookOrderChangeE2e/HookOrderChangeRepro.tsx:3]: Added a short fixture comment pointing back to #7398 so the repro is easier to find from git blame.

File: packages/example/e2e/hook-order-change.test.mts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	test('reloads the Studio when a component gains a second hook', async ({
+		page,
+	}) => {
+		const originalContent = fs.readFileSync(hookOrderChangeE2eFile, 'utf-8');
+		expect(originalContent).toContain('one hook');
+		expect(originalContent).not.toContain('React.useState(false)');
+
+		const updatedContent = originalContent.replace(
+			'\tconst frame = useCurrentFrame();\n\n\treturn <div data-frame={frame}>one hook</div>;\n',
+			`\tconst frame = useCurrentFrame();\n\tconst [enabled] = React.useState(false);\n\n\treturn <div data-frame={frame} data-enabled={enabled ? 'yes' : 'no'}>two hooks</div>;\n`,
```
[the remark this answers is not in the document]
Comment (JonnyBurger) [reply packages/example/e2e/hook-order-change.test.mts]: Removed the extra attribute and now just call the second hook directly, since the test only needs the hook order change and the rendered output proves the refresh worked.
